src/controllers/user.controller.js

```diff
@@ -162,16 +159,93 @@ class UserController {
             res,
             200,
             null,
-            'Your account has been activated.'
+            "Your account has been activated."
           );
         }
       }
 
-      return responseGenerator.sendError(res, 400, 'Invalid validation token.');
+      return responseGenerator.sendError(res, 400, "Invalid validation token.");
     } catch (error) {
       return responseGenerator.sendError(res, 500, error.message);
     }
   }
+
+  /**
+   * @static
+   * @param {object} req - express request object
+   * @param {object} res - express response object
+   * @returns {object}- returns information about the follow status of the requested user
+   * @memberof UserController
+   */
+  static async followUser(req, res) {
+    try {
+      //  Please inform team to change the id of the decrypted user to id when been parsed with req object
+      const { followeeId } = req.body;
+      const { id: followerId } = req.currentUser;
+      const details = { followeeId, followerId };
+      const followedUser = await BaseRepository.findOrCreate(
+        db.Follower,
+        details
+      );
+      const [user, created] = followedUser;
+
+      if (created) {
+        return responseGenerator.sendSuccess(
+          res,
+          200,
+          null,
+          `You just followed the user with id = ${followeeId}`
+        );
+      }
+      return responseGenerator.sendError(
+        res,
+        400,
+        `You were already following the user with id = ${followeeId}`
+      );
+    } catch (err) {
+      return responseGenerator.sendError(res, 500, err.message);
+    }
+  }
+
+  /**
+   * @static
+   * @param {object} req - express request object
+   * @param {object} res - express response object
+   * @returns {object}- returns information about the follow status of the requested user
+   * @memberof UserController
+   */
+  static async unfollowUser(req, res) {
+    try {
+      const { followeeId } = req.body;
+      const { id: followerId } = req.currentUser;
+
+      const details = {
+        followerId,
+        followeeId,
+      };
+
+      const follower = await BaseRepository.findOneByField(
+        db.Follower,
+        details
+      );
+      if (!follower) {
```

Comment:
it would be better to check if the follower returns true then send a success response then check if the user returns false or null return error message